evm_trace/base.py

```diff
@@ -1,178 +1,79 @@
-import math
-from typing import Dict, Iterator, List, Optional
+from typing import List, Optional
 
-from eth_utils import to_int
 from ethpm_types import BaseModel, HexBytes
-from pydantic import Field, validator
+from pydantic import validator
 
 from evm_trace.display import get_tree_display
 from evm_trace.enums import CallType
 
 
-class TraceFrame(BaseModel):
-    pc: int
-    op: str
-    gas: int
-    gas_cost: int = Field(alias="gasCost")
-    depth: int
-    stack: List[HexBytes] = []
-    memory: List[HexBytes] = []
-    storage: Dict[HexBytes, HexBytes] = {}
-
-    @validator("pc", "gas", "gas_cost", "depth", pre=True)
-    def validate_ints(cls, value):
-        return int(value, 16) if isinstance(value, str) else value
-
-
 class CallTreeNode(BaseModel):
+    """
+    A higher-level object modeling a node in an execution call tree.
+    Both Geth-style and Parity-style low-level data structures are
+    able to convert to this one.
+    """
+
     call_type: CallType
+    """The type of call."""
+
     address: HexBytes = HexBytes("")
+    """The contract address the call was made too."""
+
     value: int = 0
+    """The amount of value sent on the call."""
+
     depth: int = 0
+    """
+    The number of external jumps away the initially called contract (starts at 0).
+    """
+
     gas_limit: Optional[int] = None
+    """
+    The total amount of gas available.
+    """
+
     gas_cost: Optional[int] = None  # calculated from call starting and return
+    """The cost to execute this opcode."""
+
     calldata: HexBytes = HexBytes("")
+    """Transaction calldata (inputs)."""
+
     returndata: HexBytes = HexBytes("")
+    """Transaction returndata (outputs)."""
+
     calls: List["CallTreeNode"] = []
+    """The list of external sub-calls this call makes."""
+
     selfdestruct: bool = False
+    """Whether this is a SELFDESTRUCT opcode or not."""
+
     failed: bool = False
+    """Whether the call failed or not."""
 
     def __str__(self) -> str:
-        return get_tree_display(self)
+        try:
+            return get_tree_display(self)
+        except Exception as err:
+            return f"CallTreeNode (display_err={err})"
 
     def __repr__(self) -> str:
         return str(self)
 
     def __getitem__(self, index: int) -> "CallTreeNode":
         return self.calls[index]
 
+    @validator("calldata", "returndata", "address", pre=True)
+    def validate_bytes(cls, value):
+        return HexBytes(value) if isinstance(value, str) else value
 
-def get_calltree_from_geth_trace(
```

Comment:
Moved all this geth-specific stuff to its own module, similar to how we did for `parity`. Now, this base model only contains shared stuff, like `CallTreeNode`.